### PR TITLE
`display` newline

### PR DESCRIPTION
**Pull request by t-bltg**
Per https://github.com/JuliaLang/julia/pull/43787 and https://github.com/JuliaLang/julia/pull/43787#issuecomment-1013898482.

@mkitti, since I'm unfamiliar with `Compat.jl`, I hope this is correct ?

### CODE REVIEW

**Comment by fredrikekre** [comment]
It would have to be a separate function (`Compat.display`) I believe.

**Comment by mkitti** [comment]
Perhaps alternate `TextDisplay` might make sense rather than creating an alternate `display`?

**Comment by t-bltg** [comment]
My implementation doesn't seem to work.
```
ERROR: MethodError: no method matching display(::TextDisplay, ::Int64)
You may have intended to import Base.display
```

**Comment by mkitti** [comment]
I would implement a `CompatTextDisplay <: AbstractDisplay`. You could then `push!` that into `Base.Multimedia.displays` to get compatible behavior.

**Comment by t-bltg** [comment]
> I would implement a `CompatTextDisplay <: AbstractDisplay`. You could then `push!` that into `Base.Multimedia.displays` to get compatible behavior.

@fredrikekre suggested that we shouldn't override the default behavior and use `Compat.display`. Pushing to `Base.Multimedia.displays` will affect the default display, no ?

**Comment by mkitti** [comment]
The approaches are not mutually exclusive although the trick is in creating separate namespaces. You might need to create submodules to contain one or both.

The advantage of making a separate `CompatTextDisplay` is that you will be able to influence how `Base.display` works in code that you do not have access to. Perhaps you are using another module that uses `display`, and you do not want to have to vendor or modify that package directly.

Just to clear, `Compat.jl` should not necessarily `push!` `CompatTextDisplay` automatically on `__init__`. Rather this is something the end application developer might do.

File: src/Compat.jl [hunk 1]
```diff
@@ -1314,6 +1314,17 @@ if VERSION < v"1.8.0-DEV.487"
     eachsplit(str::AbstractString; limit::Integer=0, keepempty=false) =
         eachsplit(str, isspace; limit=limit, keepempty=keepempty)
 end
+
+# https://github.com/JuliaLang/julia/pull/43787
+if VERSION < v"1.8.0-DEV.1310"
+    display(d::TextDisplay, M::MIME"text/plain", @nospecialize x) = (show(d.io, M, x); println(d.io))
```
Comment (mkitti) [inline src/Compat.jl:1320]:
```suggestion
    display(d::TextDisplay, M::MIME"text/plain", @nospecialize x) = (show(d.io, M, x); println(d.io))
```

```suggestion
    import Base.Multimedia.display
    display(d::TextDisplay, M::MIME"text/plain", @nospecialize x) = (show(d.io, M, x); println(d.io))
```

**Comment by mkitti** [comment]
I just read this more carefully, and I see that now that you are mainly trying to make the new behavior available in older Julia as opposed to making the old behavior available in new Julia.

File: src/Compat.jl (hunk 1 above)
Comment (mkitti) [reply src/Compat.jl:1320]: That would be the change needed with the current approach. Without, this the current behavior of `Base.display` is not affected.

However, for a separate `Compat.display`, you would NOT want to do this.

**Comment by t-bltg** [comment]
> I just read this more carefully, and I see that now that you are mainly trying to make the new behavior available in older Julia as opposed to making the old behavior available in new Julia.

Exactly, if I'm not misunderstanding the word `latest`, quoting from `README.md`:
> the Compat package provides a macro that lets you use the latest syntax in a backwards-compatible way.

File: src/Compat.jl (hunk 1 above)
Comment (t-bltg) [reply src/Compat.jl:1320]: Why is `import Base.Multimedia.display` necessary here ?

**Comment by t-bltg** [comment]
> The advantage of making a separate `CompatTextDisplay` is that you will be able to influence how `Base.display` works in code that you do not have access to. Perhaps you are using another module that uses `display`, and you do not want to have to vendor or modify that package directly.

That might be handy.

> Just to clear, `Compat.jl` should not necessarily `push!` `CompatTextDisplay` automatically on `__init__`. Rather this is something the end application developer might do.

I'd be happy to review a PR, my implementation is not necessarily the one we need, and I don't have a clear vision of what you propose as an alternative.

**Comment by LilithHafner** [comment]
I don't think we need this anymore. IIUC breakage was all transient in scripts and the window for adding tooling with a significant impact on that has closed.

**Pull request closed without merging**